include/swift-c/DependencyScan/DSString.h

```diff
@@ -10,41 +10,41 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef SWIFT_C_DSSTRING_H
-#define SWIFT_C_DSSTRING_H
+#ifndef SWIFT_C_SWIFTSCANSTRING_H
+#define SWIFT_C_SWIFTSCANSTRING_H
 
 #include "DependencyScanMacros.h"
 
-DEPSCAN_BEGIN_DECLS
+SWIFTSCAN_BEGIN_DECLS
 
 /**
  * A character string used to pass around dependency scan result metadata.
  * A slightly-reduced variation of clang's CXString
- * Use \c ds_get_C_string() to retrieve the string data and, once finished
- * with the string data, call \c ds_dispose_string() to free the string.
+ * Use \c swiftscan_get_C_string() to retrieve the string data and, once finished
+ * with the string data, call \c swiftscan_dispose_string() to free the string.
  */
 typedef struct {
   const void *data;
   unsigned private_flags;
-} ds_string_t;
+} swiftscan_string_t;
```

Comment:
@akyrtzi do you know why CXString has this representation?  I was surprised it's not just a pointer.

Reply:
The only use of the flags seems to be to distinguishing who manages this pointer, whether it was malloc'd locally or points to unmanaged memory.

Reply:
I personally dislike `CXString`, I'd suggest to either use something like [`indexstore_string_ref_t`](https://github.com/apple/llvm-project/blob/apple/main/clang/include/indexstore/indexstore.h#L126) and be clear what is the lifetime of the returned string (e.g. "same lifetime as the object the string came from") or malloc and return a `char *` and require the caller to `free()` it.

Reply:
> be clear what is the lifetime of the returned string (e.g. "same lifetime as the object the string came from")

Note that on the Swift code you are mainly just going to copy it into a Swift string as soon as you call it, so the lifetime aspect it won't make things complicated (it will even simplify things since you won't have to call dispose on it).

Reply:
I changed the string implementation to one that resembles `index store_string_ref_t` and its lifetime is now tied to the object the string comes from. Clients need not worry about freeing string memory, it will be freed automatically when the owning object is disposed of.